rules/framework.bzl

```diff
@@ -146,9 +146,9 @@ def _concat(*args):
             arr += x
     return arr
 
-def _get_virtual_framework_info(ctx, framework_files, compilation_context_fields):
+def _get_virtual_framework_info(ctx, framework_files, compilation_context_fields, deps, transitive_deps, vfs):
     import_vfsoverlays = []
-    for dep in ctx.attr.vfs:
```

Comment:
hope we have a way to forbid us from using `ctx.attr` instead inject them in so that we don't encounter bug because of references to ctx.attr.<xxx>

Reply:
Yeah, but I don't find any way to do that.